### DISCUSSION TITLE
Inlining small images

### DISCUSSION THREAD

**Discussion by babichjacob** [JonasKruckenberg/imagetools#85] [Ideas]
This seems to be the first mention of inlining images from / through `imagetools`. Is this a feature that currently exists or that we can help bring into existence? If so -- do you think it's already possible with [custom directive](https://github.com/JonasKruckenberg/imagetools/blob/main/docs/guide/extending.md#custom-directive) or [custom output format](https://github.com/JonasKruckenberg/imagetools/blob/main/docs/guide/extending.md#custom-output-formats) support that `imagetools` already has or that changes need to be made to one of the first party packages to support it?

Specifically as it concerns Vite, there is a [`build.assetsInlineLimit` option](https://vitejs.dev/config/#build-assetsinlinelimit) that can be used without adding a new option / directive.

But Rollup doesn't have built-in asset handling (?), so there'd have to be an extra option added to `rollup-plugin-imagetools`.

Thanks!

**Comment by JonasKruckenberg** [comment]
This is not yet possible with customDirectives/customOutputFormats, however the need for small plcaeholder images [has come up before](https://github.com/JonasKruckenberg/imagetools/discussions/69) which is somewhat related. This is definitely something I want to get implemented!

I would imagine this idea needing additional options like the ones [@rollup/plugin-url](https://github.com/rollup/plugins/tree/master/packages/url/#readme) has so it could work in `rollup-plugin-imagetools`.

I'm currently working on a refactor of the metadata handling that would enable this to be addressed by a custom output-format as well!

**Comment by rchrdnsh** [comment]
This would be very cool, and might be nice for the whole 'blur up' thing, which I personally quite like...when done right, of course. So I look forward to this :-)
